api/lib/domain/services/schooling-registrations-xml-service.js

```diff
@@ -1,91 +1,210 @@
+      return reject_(e);
+    };
+
+    const mapSchoolingRegistrationsByStudentId = new Map();
+    const nationalStudentIds = [];
+
+    const streamerToParseSchoolingRegistrations = new saxPath.SaXPath(saxParser, NODES_SCHOOLING_REGISTRATIONS);
+    streamerToParseSchoolingRegistrations.on('match', (xmlNode) => {
+      if (_isSchoolingRegistrationNode(xmlNode)) {
+        xml2js.parseString(xmlNode, (err, nodeData) => {
+          try {
+            if (err) throw err;
+            processStudentsNodes(mapSchoolingRegistrationsByStudentId, nodeData, nationalStudentIds);
+            processStudentsStructureNodes(mapSchoolingRegistrationsByStudentId, nodeData);
+          } catch (err) {
+            reject(err);
+          }
+        });
+      }
+    });
+
+    siecleFileStream.pipe(saxParser);
+
+    streamerToParseSchoolingRegistrations.on('end', () => {
+      resolve(Array.from(mapSchoolingRegistrationsByStudentId.values()));
+    });
+  });
 }
 
-function _filterSchoolingRegistrationsWithNoDivision(xmlSchoolingRegistration, xmlSchoolingRegistrationsStructures) {
-  const schoolingRegistrationStructuresContainer = _findSchoolingRegistrationStructures(xmlSchoolingRegistration, xmlSchoolingRegistrationsStructures);
-  const schoolingRegistrationDivision = _findSchoolingRegistrationDivision(schoolingRegistrationStructuresContainer);
-  return !_.isEmpty(schoolingRegistrationDivision) && schoolingRegistrationDivision !== 'Inactifs';
+function _mapStudentInformationToSchoolingRegistration(nodeData) {
+  return {
+    lastName: _getValueFromParsedElement(nodeData.ELEVE.NOM_DE_FAMILLE),
+    preferredLastName: _getValueFromParsedElement(nodeData.ELEVE.NOM_USAGE),
+    firstName: _getValueFromParsedElement(nodeData.ELEVE.PRENOM),
+    middleName: _getValueFromParsedElement(nodeData.ELEVE.PRENOM2),
+    thirdName: _getValueFromParsedElement(nodeData.ELEVE.PRENOM3),
+    birthdate: moment(nodeData.ELEVE.DATE_NAISS, 'DD/MM/YYYY').format('YYYY-MM-DD') || null,
+    birthCountryCode: _getValueFromParsedElement(nodeData.ELEVE.CODE_PAYS, null),
+    birthProvinceCode: _getValueFromParsedElement(nodeData.ELEVE.CODE_DEPARTEMENT_NAISS),
+    birthCityCode: _getValueFromParsedElement(nodeData.ELEVE.CODE_COMMUNE_INSEE_NAISS),
+    birthCity: _getValueFromParsedElement(nodeData.ELEVE.VILLE_NAISS),
+    MEFCode: _getValueFromParsedElement(nodeData.ELEVE.CODE_MEF),
+    status: _getValueFromParsedElement(nodeData.ELEVE.CODE_STATUT),
+    nationalStudentId: _getValueFromParsedElement(nodeData.ELEVE.ID_NATIONAL),
+  };
 }
 
-function _filterNotYetArrivedSchoolingRegistrations(xmlSchoolingRegistration) {
-  const nationalSchoolingRegistrationId = _getEitherElementValueOrNull(xmlSchoolingRegistration, 'ID_NATIONAL');
-  return !_.isEmpty(nationalSchoolingRegistrationId);
+async function _readFirstLineFromFile(path) {
+  return new Promise((resolve, reject) => {
+    const readStream = fs.createReadStream(path);
+    const lineEnding = '\n';
```

Comment:
Permet de souligner le fait que l'on recherche un caractère, pas une string
```suggestion
    const lineEndingCharacter = '\n';
```